tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,99 @@
+# Weight clustering
+
+This document provides an overview on weight clustering to help you determine how it fits with your use case. To dive right into the code, see the [weight clustering end-to-end example](clustering_example.ipynb) and the [API docs](../../api_docs/python). For additional details on how to use the Keras API, a deep dive into weight clustering, and documentation on more advanced usage patterns, see the [weight clustering comprehensive guide](clustering_comprehensive_guide.ipynb).
+
+## Overview
+
+Clustering, or weight sharing, reduces the number of unique weight values in a model, leading to benefits for deployment. It first groups the weights of each layer into *N* clusters, then shares the cluster's centroid value for all the weights belonging to the cluster.
+
+This technique brings improvements in terms of model compression. By reducing the number of unique weight values, weight clustering renders the weights suitable for compression via Huffman coding and similar techniques. Future framework support will, therefore, be able to reduce memory size and bandwidth requirements. This can be critical for deploying deep learning models on embedded systems with limited resources.
+
+We have seen up to 5x improvements in model compression with minimal loss of accuracy, as demonstrated by the [results](#results) presented below. The compression gains depend on the model and the accuracy targets in each specific use case. For example, for the MobileNetV2 image classification model, one can choose to reduce all non-depthwise convolutional layers to use just 32 unique weigth values and obtain a float32 tflite model that is approximately 4.8 times more compressible using ZIP Deflate algorithm than the original model. However, that will result in about 3% drop of the top-1 classification accuracy. On the other hand, the same model clustered less agressively, using 256 clusters for two internal layers and 32 clusters for the final convolutional layer, maintains virtually the same accuracy as the original model, yet still yields a respectable 1.8x improvement in compression ratio.
```

Comment:
On "the compression gains depend on the model and the accuracy targets in each specific use case. ... yet still yields a respectable 1.8x improvement in compression ratio."

I'd update the table based on what you did in the RFC to provide these details, and then remove the paragraph. The table gets the message across better.